Spokestack/NLUTensorflowTokenizer.swift

```diff
@@ -82,31 +82,33 @@ internal struct WordpieceTokenizer: Tokenizer {
         }
     }
     
-    /// Decode and detokenize the encoded tokens into a space-separated string.
-    /// - Parameter tokens: The encoded tokens to decode and detokenize.
-    func detokenize(_ tokens: [String]) throws -> String {
-        return tokens.reduce("", { (result, s) in
+    func decode(_ tokens: [String]) ->  [String] {
+        return tokens
+            .compactMap({ $0.trimmingCharacters(in: .whitespacesAndNewlines) })
+            .reduce([], { (result, s) in
             if s.prefix(2) == self.piecePrefix {
-                return result + s.dropFirst(2)
+                return result.dropLast() + [(result.last ?? "") + String(s.dropFirst(2))]
             } else if result.count > 0 {
-                return result + " " + s
+                return result + [s]
             } else {
-                return s
+                return [s]
             }
         })
     }
+    
+    /// Decode and detokenize the encoded tokens into a space-separated string.
+    /// - Parameter tokens: The encoded tokens to decode and detokenize.
+    func detokenize(_ tokens: [String]) -> String {
+        return self.decode(tokens).joined(separator: " ")
```

Comment:
The behavior is partially documented now, but the unconditional space separation and failure to retain the original text will still hurt display of the slot values.